src/Bicep.Core/Configuration/bicepconfig.json

```diff
@@ -40,6 +40,18 @@
       }
     }
   },
+  "providers": {
+    "az": {
+      "builtIn": true
+    },
+    "kubernetes": {
+      "builtIn": true
+    },
+    "microsoftGraph": {
+      "builtIn": true
+    }
+  },
+  "implicitProviders": ["az"],

```

Comment:
I thought that at the last team discussion, we landed on something like:
```json
{
  ...
  "providerSources": {
    "az": "builtin",
    "kubernetes": "builtin",
    "microsoftGraph": "builtin"
  },
  "implicitProviders": ["az"],
  ...
}
```

Reply:
Yes, eventually we will move to this representation, however, this requires a REP amendment. I will address this change in a separate PR.

Reply:
Also, should we call them `extensionSources` and `implicitExtensions` to avoid breaking changes later on, because we plan to stop using the term `provider`?

Reply:
> Yes, eventually we will move to this representation, however, this requires a REP amendment. I will address this change in a separate PR.

In that case do you want to hold off on this PR until after the 0.26 release? It's coming up fairly soon, and I'm not sure we want to ship something we know we will quickly make breaking changes to.

Reply:
Btw, I think we can always update the REP later at some point to reflect the implementation. We need to make the changes before the next Bicep release.

Reply:
The rep is already waiting for review here: https://github.com/Azure/bicep-reps/pull/5
I don't think we need to wait because it's an experimental feature and it's hidden behind a feature flag. Can you explain why hold?

Reply:
> I don't think we need to wait because it's an experimental feature and it's hidden behind a feature flag. Can you explain why hold?

It's one thing to release something that we think will have breaking changes in the future, but it feels counterproductive to release something we **know** we will immediately make breaking changes to, especially when we already know what breaking changes we will be making. Releasing the version described in https://github.com/Azure/bicep-reps/pull/5 in 0.26 would be fine.

Reply:
In this case we are not releasing anything. This whole functionality is under an experimental feature flag. There are no guarantees of backwards compatibility or non-breaking changes on capabilities that are only available when the feature flag is set.

My hope is that I can get this relatively large PR merged so that I don't keep wasting time resolving conflicts each time. The design in  https://github.com/Azure/bicep-reps/pull/5 is not yet approved and has caveats and drawbacks that need to be considered compared to the current design. The current design is the plan of record and so I think is ok to merge it until we decide that a better alternative is found.

For now I am resolving this with the commitment to change the code in a separate PR once the design in https://github.com/Azure/bicep-reps/pull/5 is finalized and approved.